Process.c

```diff
@@ -1120,3 +1120,74 @@ int Process_compareByKey_Base(const Process* p1, const Process* p2, ProcessField
       return SPACESHIP_NUMBER(p1->pid, p2->pid);
    }
 }
+
+void Process_updateComm(Process* this, const char* comm) {
+   if (!this->procComm && !comm)
+      return;
+
+   if (this->procComm && comm && String_eq(this->procComm, comm))
+      return;
+
+   free(this->procComm);
+   this->procComm = comm ? xStrdup(comm) : NULL;
+   this->mergedCommand.commChanged = true;
+}
+
+static int skipPotentialPath(const char* cmdline, int end) {
+   if (cmdline[0] != '/')
+      return 0;
+
+   int slash = 0;
+   for (int i = 1; i < end; i++) {
+      if (cmdline[i] == '/' && cmdline[i+1] != '\0') {
+         slash = i + 1;
+         continue;
+      }
+
+      if (cmdline[i] == ' ' && cmdline[i-1] != '\\')
+         return slash;
+
+      if (cmdline[i] == ':' && cmdline[i+1] == ' ')
+         return slash;
+   }
+
+   return slash;
+}
+
+void Process_updateCmdline(Process* this, const char* cmdline, int basenameStart, int basenameEnd) {
+   assert(basenameStart >= 0);
+   assert((cmdline && basenameStart < (int)strlen(cmdline)) || (!cmdline && basenameStart == 0));
+   assert(basenameEnd >= 0);
```

Comment:
Maybe even …
```suggestion
   assert(basenameEnd >= basenameStart);
```

Reply:
This does not work for the following process:

*/proc/2099/stat*:  `2099 (file.so) S ...`
*/proc/2099/exe* -> */usr/bin/kdeinit5*
cmdline: 
```
file.so
[kdeinit5]
file
local:/run/user/1000/klauncheriCClUH.1.slave-socket
local:/run/user/1000/kded5LIpHLP.1.slave-socket'
```
resulting in `start=45 end=7` at https://github.com/BenBE/htop/blob/2566737fe138ea474e4fd834267b8341db5ef560/linux/LinuxProcessList.c#L1029-L1131

Reply:
Please check the Linux implementation for the colon check, as the one used there is tested to work with both wine and some other cases, thus basically reconstructing the heuristic from the Linux platform side.

Reply:
The Linux heuristic, the code I quoted, results in start > end.

Reply:
Ah, k. Then this is fine, as the existing code already had to handle this situation.